tests/core/test_units.py

```diff
@@ -94,11 +94,11 @@ def test_length(self):
         assert str(y.unit) == "ang^3"
 
     def test_memory(self):
-        mega = Memory(1, "Mb")
+        mega = Memory(1, "MB")
         assert mega.to("byte") == 1024**2
         assert mega == Memory(1, "mb")
 
-        same_mega = Memory.from_str("1Mb")
+        same_mega = Memory.from_str("1MB")
```

Comment:
For some reason `Memory.from_str("1 MB")` failed to detect `unit_type`, which works for other units.

As per the [SI Unit rules and style conventions](https://physics.nist.gov/cuu/Units/checklist.html):
> There is a space between the numerical value and unit symbol, even when the value is used in an adjectival sense, except in the case of superscript units for plane angle.